pydantic/_internal/_validators.py

```diff
@@ -226,3 +226,55 @@ def ip_v6_interface_validator(__input_value: Any) -> IPv6Interface:
         return IPv6Interface(__input_value)
     except ValueError:
         raise PydanticCustomError('ip_v6_interface', 'Input is not a valid IPv6 interface')
+
+
+def greater_than_validator(x: Any, gt: Any) -> Any:
+    if not (x > gt):
+        raise PydanticCustomError('greater_than', 'Input should be greater than {gt}', {'gt': gt})
```

Comment:
```suggestion
        raise PydanticKnownError('greater_than', {'gt': gt})
```

Reply:
Won't work:

```
    def greater_than_validator(x: Any, gt: Any) -> Any:
        if not (x > gt):
>           raise PydanticKnownError('greater_than', {'gt': gt})
E           TypeError: GreaterThan: 'gt' context value must be a Number
```

Reply:
Should be fixed in core, bit this is fine for now.

Reply:
I hadn't realised it was every single one that was using `PydanticCustomError`, we should adopt `PydanticKnownError` where possible.

Reply:
Unless I’m missing something, you already pointed that out and as per above, it won’t work unless we make extensive changes in pydantic-core. It also makes no difference for users as far as I can tell.